### PR TITLE
[CPU]Fix coverity scan issues

### PR DESCRIPTION
**Pull request by chenhu-wang**
### Details:
 - *Overflowed integer argument (INTEGER_OVERFLOW)*

### Tickets:
 - *CVS-153060*

### CODE REVIEW

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp [hunk 1]
```diff
@@ -2835,11 +2835,12 @@ void Interpolate::InterpolateJitExecutor::pillowCGathered(const uint8_t *in_ptr_
             size_t threadsNum = parallel_get_num_threads();
             size_t parallelNum = B;
             // IH * OW * C buf needed
+            size_t buffer_size = static_cast<size_t>(OW * IH * C);
             if (parallelNum < threadsNum) {
-                arg.src_ptr[1] = static_cast<uint8_t*>(&pillow_working_buf[(b * OW * IH * C) * srcDataSize]);
+                arg.src_ptr[1] = static_cast<uint8_t*>(&pillow_working_buf[b * buffer_size * srcDataSize]);
```
[the remark this answers is not in the document]
Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp:2840]: I think `uint8_t *in_ptr_`  is allowed to add a negative value on line 2833, but index inside `pillow_working_buf[index]` can not allowed a negative value from scan perspective. (OW, IH and C is type of int and could be negative)

[the remark this answers is not in the document]
Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp:2840]: The log on scan tool:
CID 1559426: (#1 of 1): Overflowed integer argument (INTEGER_OVERFLOW)
10. overflow_sink: threadsIdx * this->OW * this->IH * this->C * this->this->srcDataSize, which might have underflowed, is passed to this->this->pillow_working_buf[threadsIdx * this->OW * this->IH * this->C * this->this->srcDataSize]. [Note: The source code implementation of the function has been overridden by a builtin model.]
2842                arg.src_ptr[1] = static_cast<uint8_t*>(&pillow_working_buf[(threadsIdx * OW * IH * C) * srcDataSize]);

[the remark this answers is not in the document]
Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp:2840]: The type promotion rule for `size_t * int` is, operands will be converted(promote) to size_t implicit if one is not this type, then perform compute. "int convert to size_t risk" inside array[index] is not allowed in the scan. But this risk is allowed in value of  `uint8_t *in_ptr_ + value`. Interesting scan rule:-)

[the remark this answers is not in the document]
Comment (dmitry-gorokhov) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp:2840]: Ok. I see. Thanks for the explanation.

**Pull request merged**